### PR TITLE
Auto labels for PR's and issues

### PR DESCRIPTION
**Pull request by HGWright**
### Description

This PR adds an auto labeling gha that will label both issues and PR's. It does this by either looking at the files changed or the text within the titles and body of the issue/PR. If it finds certain predefined words it will add the corresponding label.

I have made a couple of extra labels for the repo as I felt they were missing from the current list. part of #106 

I have tested this on my personal fork and it seems to work as expected. It runs on the opening, reopening or editing of the issue/pr.

I think a follow on from this PR is to think about updating the templates for issues and PR's. I have not done this here but am interested in your thoughts on my current thoughts about this.

Currently I think not updating them is the best option and maybe expand the list of words so that most people will accidently trigger the correct labels. The whole point is that they are supposed to be automated so making the auther tailor their PR to the labeller defeats the point of the automation.

I think expanding the list of words is probably necessary for this because the reference words are case sensitive so this could make them not trigger with we would want them too

---

### CODE REVIEW

File: .github/workflows/ci-label.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+on:
+  pull_request_target:
+    types: [opened, edited, reopened]
+  issues:
+    types: [opened, edited, reopened]
+
+permissions:
+  issues: write
+  pull-requests: write
+  contents: read
```
Comment (bjlittle) [inline .github/workflows/ci-label.yml:16]: @HGWright From what I understand the default permissions for the GHA `GITHUB_TOKEN` is [permissive](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token).

So, just clarifying for my understanding... for `issues` and `pull-requests` you're not changing the permissions, but for `contents` you're restricting to `read` only, is that right?

Also, what's the reason for restricting the `contents` permissions? What's the implications?

**Review by bjlittle — CHANGES_REQUESTED** [review]
@HGWright This looks great! :100: 

Just a simple question to clarify, then we can bank this and start auto-labeling :tada:

File: .github/workflows/ci-label.yml (hunk 1 above)
Comment (HGWright) [reply .github/workflows/ci-label.yml:16]: You are right @bjlittle, although the permissions were not set up by default on my fork. So I thought I needed them to make it work. I will remove them from the workflow.

As far as the `contents` is concerned to be honest it was with the others in the READ.ME for the issue-labeler action so I added it with the others. Off the top of my head the only thing I can think of is that the labeler would never need to write to the contents of a pr/issue, so I guess I can remove that too.

Comment (bjlittle) [reply .github/workflows/ci-label.yml]: I guess if less permissive permissions work, then just roll with that. Happy either way 👍

**Review by bjlittle — APPROVED** [review]

**Pull request merged**